From f96601ce2b8b3cba890fe13626e800f502366ca0 Mon Sep 17 00:00:00 2001
From: snek <snek@chromium.org>
Date: Tue, 20 Aug 2024 07:23:58 -0700
Subject: [PATCH] [maglev] fix invalid near jump

Fixes this CHECK failure:
> Fatal error in src/codegen/x64/assembler-x64.cc
> Check failed: is_int8(disp).

Change-Id: I9ad3a5a1504d1e06080b9e0653b963061e19c720
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/5797736
Commit-Queue: snek <snek@chromium.org>
Reviewed-by: Leszek Swirski <leszeks@chromium.org>
Cr-Commit-Position: refs/heads/main@{#95726}
---
 src/maglev/maglev-ir.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/maglev/maglev-ir.cc b/src/maglev/maglev-ir.cc
index 1ddd3c11a5d..c39c7e6cd75 100644
--- a/src/maglev/maglev-ir.cc
+++ b/src/maglev/maglev-ir.cc
@@ -1721,7 +1721,7 @@ void TryUnboxNumberOrOddball(MaglevAssembler* masm, DoubleRegister dst,
   // If Smi, convert to Float64.
   __ SmiToInt32(clobbered_src);
   __ Int32ToDouble(dst, clobbered_src);
-  __ Jump(&done, Label::kNear);
+  __ Jump(&done);
   __ bind(&is_not_smi);
   JumpToFailIfNotHeapNumberOrOddball(masm, clobbered_src, conversion_type,
                                      fail);
-- 
2.46.0

